packages/fast-usdc/contract/quickSend.flows.js

```diff
@@ -0,0 +1,185 @@
+import { mustMatch } from '@agoric/internal';
+import { atob } from '@endo/base64';
+import { M } from '@endo/patterns';
+import { ChainAddressShape } from '@agoric/orchestration';
+import {
+  AgoricCalc,
+  NobleCalc,
+} from '@agoric/orchestration/src/utils/address.js';
+
+/**
+ * @import {ExecutionContext} from 'ava';
+ *
+ * @import {Passable} from '@endo/pass-style';
+ * @import {Guarded} from '@endo/exo';
+ * @import {GuestInterface} from '@agoric/async-flow';
+ * @import {ChainAddress, OrchestrationAccountI, OrchestrationFlow, Orchestrator, ZcfTools} from '@agoric/orchestration';
+ * @import {VTransferIBCEvent} from '@agoric/vats';
+ * @import {ResolvedContinuingOfferResult} from '@agoric/orchestration/src/utils/zoe-tools.js';
+ * @import {InvitationMakers} from '@agoric/smart-wallet/src/types.js';
+ * @import {QuickSendTerms} from './quickSend.contract.js';
+ * @import {FungibleTokenPacketData} from '@agoric/cosmic-proto/ibc/applications/transfer/v2/packet.js';
+ * @import {TypedPattern} from '@agoric/internal';
+ * @import {ChainHub} from '@agoric/orchestration';
+ */
+
+const AddressShape = M.string(); // XXX
+
+/**
+ * @typedef {{
+ *   amount: NatValue;
+ *   dest: ChainAddress;
+ *   nobleFwd: string;
+ * }} CallDetails
+ */
+
+/** @type {TypedPattern<CallDetails>} */
+const CallDetailsShape = harden({
+  amount: M.nat(),
+  dest: ChainAddressShape,
+  nobleFwd: AddressShape,
+});
+
+const { add, make, subtract } = AmountMath;
+
+/**
+ * @typedef {{
+ *   settlement: OrchestrationAccountI;
+ *   fundingPool: OrchestrationAccountI;
+ *   feeAccount: OrchestrationAccountI;
+ * }} QuickSendAccounts
+ */
+
+/**
+ * @satisfies {OrchestrationFlow}
+ * @param {Orchestrator} orch
+ * @param {{
+ *   terms: QuickSendTerms & StandardTerms;
+ *   chainHub: GuestInterface<ChainHub>;
+ *   t?: ExecutionContext<{ nextLabel: Function }>; // XXX
+ *   makeSettleTap: (accts: QuickSendAccounts) => Guarded<{
+ *     receiveUpcall: (event: VTransferIBCEvent) => void;
+ *   }>;
+ *   makeWatcherCont: (accts: QuickSendAccounts) => {
+ *     invitationMakers: InvitationMakers;
+ *     actions: Record<string, Function>;
+ *   };
+ * }} ctx
+ * @param {ZCFSeat} seat
+ * @param {{}} _offerArgs
+ */
+export const initAccounts = async (orch, ctx, seat, _offerArgs) => {
+  const { log = console.log } = ctx.t || {};
+
+  const agoric = await orch.getChain('agoric');
+  const aInfo = await agoric.getVBankAssetInfo();
+  for (const a of aInfo) {
+    if (a.brand === ctx.terms.brands.USDC) {
+      const [baseName, baseDenom] = ['noble', 'uusdc']; // ???
+      await orch.getChain(baseName);
+      ctx.chainHub.registerAsset(a.denom, {
+        baseDenom,
+        baseName,
+        chainName: 'agoric',
+        brand: ctx.terms.brands.USDC,
+      });
+      break;
+    }
+  }
+
+  const fundingPool = await agoric.makeAccount();
+  const settlement = await agoric.makeAccount();
+  const feeAccount = await agoric.makeAccount();
```

Comment:
Can fee account be a `Purse`? Only `fundingPool` and `settlement` might need to send or accept IBC transfers?

Reply:
In recent discussion, I confirmed with @sufyaankhan that it should be a normal cosmos account that the fee beneficiary creates before the contract launches; that way, they can withdraw at will using normal `tx bank send`.

But I suppose it could be a `Seat`, yes. (a Purse would give the contract direct access; we avoid that, normally)